src/engine/TransitivePathHashMap.cpp

```diff
@@ -3,31 +3,89 @@
 // Author: Florian Kramer (florian.kramer@neptun.uni-freiburg.de)
 //         Johannes Herrmann (johannes.r.herrmann(at)gmail.com)
 
-#include "TransitivePathHashMap.h"
+#include "engine/TransitivePathHashMap.h"
 
 #include <memory>
 
 #include "engine/CallFixedSize.h"
 #include "engine/TransitivePathBase.h"
 
 // _____________________________________________________________________________
-HashMapWrapper TransitivePathHashMap::setupEdgesMap(
-    const IdTable& dynSub, const TransitivePathSide& startSide,
-    const TransitivePathSide& targetSide) const {
-  return CALL_FIXED_SIZE((std::array{dynSub.numColumns()}),
-                         &TransitivePathHashMap::setupEdgesMap, this, dynSub,
-                         startSide, targetSide);
+HashMapWrapper::HashMapWrapper(
+    Map map, const ad_utility::AllocatorWithLimit<Id>& allocator)
+    : graphMap_{allocator},
+      map_{nullptr},
+      emptySet_{allocator},
+      emptyMap_{allocator} {
+  graphMap_.try_emplace(Id::makeUndefined(), std::move(map));
+  map_ = &graphMap_.at(Id::makeUndefined());
+}
+
+// _____________________________________________________________________________
+HashMapWrapper::HashMapWrapper(
+    MapOfMaps graphMap, const ad_utility::AllocatorWithLimit<Id>& allocator)
+    : graphMap_{std::move(graphMap)},
+      map_{&emptyMap_},
+      emptySet_{allocator},
+      emptyMap_{allocator} {}
+
+// _____________________________________________________________________________
+const Set& HashMapWrapper::successors(const Id node) const {
+  auto iterator = map_->find(node);
+  if (iterator == map_->end()) {
+    return emptySet_;
+  }
+  return iterator->second;
+}
+
+// _____________________________________________________________________________
+absl::InlinedVector<std::pair<Id, Id>, 1>
+HashMapWrapper::getEquivalentIdAndMatchingGraphs(Id node) const {
+  absl::InlinedVector<std::pair<Id, Id>, 1> result;
+  for (const auto& [graph, map] : graphMap_) {
+    if (node.isUndefined()) {
```

Comment:
The special meaning of undefined definitely needs some documentation.... (same as above).